### PR TITLE
Fix: UI not showing unmarshal errors

### LINKED ISSUE #2273: An error loading the manifest while the deploy command is executed doesn't contain value for the stage property on json output
**Describe the bug**
When I use the output flag to specify `json`, if there is an error reading the manifest, that error is logged without any section, so it is not displayed in the UI. We should review all the places where we are not setting any `stage`

If you see this screenshot, you can see that the pipeline failed but the logs don't show any error. Checking the pod logs, you see:

```
{"level":"error","stage":"","message":" x  Path '/***/src/non-existent-folder' does not exist","timestamp":1645360958}
{"level":"info","stage":"","message":"    Update the `sync` field in your *** manifest file to a valid directory path.","timestamp":1645360958}
{"level":"info","stage":"","timestamp":1645360958,"message":"\nFAILED: okteto pipeline has failed with: exit status 1."}
{"level":"info","stage":"","timestamp":1645360958,"message":"See https://okteto.com/docs/cloud/okteto-pipeline for details on how to configure your okteto pipeline."}
{"level":"error","stage":"","timestamp":1645360958,"message":"\u001b[91mexit status 1\u001b[0m"}
{"level":"info","stage":"done","timestamp":1645360958,"message":"EOF"}
```

Captura de pantalla 2022-02-20 a las 13 45 58

**To Reproduce**
Deploy this repository `https://github.com/ifbyol/movies` in Okteto Cloud or Staging

**Expected behavior**
The error should be displayed in the UI so user is able to find out the real issue

**Desktop (please complete the following information):**
-

**Additional context**
-

### PR DESCRIPTION
**Pull request by jLopezbarb**
Fixes #2273

## Proposed changes
- If there is an error reading the manifest context fallback to read context  
- Set the initial stage to log possible errors on the UI

### CODE REVIEW

**Review by pchico83 — APPROVED** [review]

**Comment by pchico83** [comment]
@jLopezbarb could you attach a screenshot of how the UI looks like for a manifest serialization error?

**Comment by pchico83** [comment]
@jLopezbarb what about if the error happens after the execution of the commands? for example, when updating the configmap?

**Comment by jLopezbarb** [comment]
> @jLopezbarb could you attach a screenshot of how the UI looks like for a manifest serialization error?
This is how it will like 
Captura de pantalla 2022-02-28 a las 13 46 43



> @jLopezbarb what about if the error happens after the execution of the commands? for example, when updating the configmap?

It will be shown on the CLI but not in the UI, we decided to do this also with showing endpoints. Should we create another step to show this info?

**Comment by pchico83** [comment]
@jLopezbarb that would be an option yes

**Comment by ifbyol** [comment]
Another option could be to have a general "error" or "internal server error" stage for those errors happening outside a stage. The fact of not displaying messages without stages in UI is a bit scary and could hide errors. So I think that adding that "error" stage to any final error thrown in the CLI (that is not within a stage) would help us to display those errors in the UI always

**Comment by rberrelleza** [comment]
I think we should show this error in the UI more prominently. Given that we have new manifest format coming soon, having clear errors will cut down on the support questions.

Also, this shouldn't be an 'internal server error', since it's caused by the user (in case we display a category or something )

**Comment by pchico83** [comment]
@jLopezbarb let's create a stage for the manifest deserialization

**Comment by jLopezbarb** [comment]
This is how it looks like the new stage
Captura de pantalla 2022-03-01 a las 12 28 12

**Comment by ifbyol** [comment]
> I think we should show this error in the UI more prominently. Given that we have new manifest format coming soon, having clear errors will cut down on the support questions.
> 
> Also, this shouldn't be an 'internal server error', since it's caused by the user (in case we display a category or something )

I meant the possible errors related with the configmap update or things like that. Not the specific for the manifest. The problem is that if we don't specify a stage in the logs and there is an error, it won't appear in the UI, so it would be very confusing for users.

I agree that having specific errors for the manifest is very meaningful the users and I think having is own stage is really useful

**Comment by pchico83** [comment]
> > I think we should show this error in the UI more prominently. Given that we have new manifest format coming soon, having clear errors will cut down on the support questions.
> > Also, this shouldn't be an 'internal server error', since it's caused by the user (in case we display a category or something )
> 
> I meant the possible errors related with the configmap update or things like that. Not the specific for the manifest. The problem is that if we don't specify a stage in the logs and there is an error, it won't appear in the UI, so it would be very confusing for users.
> 
> I agree that having specific errors for the manifest is very meaningful the users and I think having is own stage is really useful

The configmap one is specially tricky because logs won't be stored in the configmap either.

File: cmd/deploy/deploy.go [hunk 1]
```diff
@@ -206,10 +208,12 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 		return err
 	}
 	var err error
+	oktetoLog.SetStage("Read manifest")
 	deployOptions.Manifest, err = dc.GetManifest(deployOptions.ManifestPath)
 	if err != nil {
 		return err
 	}
+	oktetoLog.SetStage("")
```
[the remark this answers is not in the document]
Comment (ifbyol) [reply cmd/deploy/deploy.go:216]: So, are we going to include this in the `1.15.x` and do backend release with this fix?

**Comment by pchico83** [comment]
LGTM, adding two comments:
- The error about not having deploy commands should be a `Read manifest` stage error
- We need to show a custom UI for `Read manifest` and `Internal server error` stages

**Comment by maroshii** [comment]
> is there a way to hide the Run from the beginning of a command?

We can. We need to consider it a _special step_ as we do for setup

**Comment by rberrelleza** [comment]
@pchico83 @jLopezbarb sorry for the late feedback, but  "Read manifest" sounds a bit odd. Is there prior art on this? I think we shouldn't use `Run "read manifest"` but instead directly name the step "Load Manifest" (without the run) to separate more clearly between user-defined steps VS okteto defined steps.

**Comment by jLopezbarb** [comment]
I agree on naming it "Load manifest". I think we all agree on not showing Run at the beginning for the "Load manifest" step and also for the internal errors

**Comment by ifbyol** [comment]
> I agree on naming it "Load manifest". I think we all agree on not showing Run at the beginning for the "Load manifest" step and also for the internal errors

that has to be done in the app project, @jLopezbarb can you open a ticket there, please?

**Comment by rlamana** [comment]
+1 to "Load manifest"

**Pull request merged**